### PR TITLE
Added a video model, and one smoke test for it

### PR DESCRIPTION
**Pull request by ethanpailes**

### CODE REVIEW

File: app/src/ts/video-model.ts [hunk 1]
```diff
@@ -1,26 +1,228 @@
...
+                         Promise.resolve(localVideoFile),
+                         commentReplies, videoReplies);
+    }
+
+    // @returns a promise for a dataID pointing to the written video meta-node
+    public async write(): Promise<DataIDHandle> {
+        await this.commentReplies.save().catch(err => {
+            // if the appendable data already exists, ignore the error. We don't need
+            // to update it.
+            if ((typeof err.res.statusCode !== "undefined" && err.res.statusCode === 400)
```
Comment (GrahamGoudeau) [inline app/src/ts/video-model.ts]: These comparisons will work fine if you just do `err.res.statusCode === 400`.  If we did still want to check for existence too, I think that `err.res.statusCode == null` is the idiomatic way to do that.

File: app/spec/test-util.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+export const TEST_DATA_DIR: string = `${__dirname}/../../../spec/test-data`;
+
+export function failDone<T>(promise: Promise<T>,
+                            done: () => void): Promise<T> {
+    return promise.catch((err) => {
+        console.error(JSON.stringify(err));
+        fail(err); done(); throw err;
```
Comment (GrahamGoudeau) [inline app/spec/test-util.ts:8]: How is `fail()` in scope here?

Comment (ethanpailes) [reply app/spec/test-util.ts:8]: It's a magic jasmine function. I think its a global.

File: app/src/ts/video-model.ts (hunk 1 above)
Comment (ethanpailes) [reply app/src/ts/video-model.ts]: I think I should change the first checks to (typeof err.res !== "undefined")

**Pull request merged**